# Data package for transcriptomic profiles from yellow fever infected cells

## Sources

  * Experimental data were generated by Roche M, Mass√©-Deragon N, Barban V, and Legras-Lachuer C.
    * Citation is not yet available
    * GEO data set: [GSE99081](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE99081)
  * Processing:
    * Sequencing reads were downloaded from SRA, at [PRJNA387233](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA387233)
    * `fastq` files were check for adapter content, using FASTQC (no adapter sequence was found)
    *  Reads were aligned on mouse genome GRCh38.p10 + 92 ERCC sequences, with Gencode v27 annotation, using STAR 2.7.1a, and then quantified by RSEM 1.3.0 for abundance levels at gene and transcript levels.
  
## Usage

Install the package, import the library and load the data set

```R
devtools::install_github('ttdtrang/data-rnaseq-YFV')
library(data.rnaseq.YFV)
data(yfv.rnaseq.gene)
dim(yfv.rnaseq.gene@assayData$exprs)
```

The package includes 2 data sets, one for transcript-level counts/TPM and another for gene-level counts/TPM. Counts are non-integer estimate of `expected_count` by RSEM.

## Steps to re-produce data curation

1. `cd data-raw`
2. Download all necessary raw data files which include
```
8.7M	feature_attrs.transcripts.tsv
2.1M	matrix.gene.expected_count.RDS
2.0M	matrix.gene.tpm.RDS
7.8M	matrix.transcripts.expected_count.RDS
6.4M	matrix.transcripts.tpm.RDS
16K	PRJNA387233_metadata_cleaned.tsv
8.0K	starLog.final.tsv
```
3. Set the environment variable `DBDIR` to point to the path containing said files
4. Run the R notebook `make-data-package.Rmd` to assemble parts into `ExpressionSet` objects.

You may need to change some code chunk setting from `eval=FALSE` to `eval=TRUE` to make sure all chunks would be run. These chunks are disabled by default to avoid overwriting existing data files in the folder.
